pkg/envoy/ads/server.go

```diff
@@ -26,7 +27,7 @@ const ServerType = "ADS"
 func NewADSServer(meshCatalog catalog.MeshCataloger, enableDebug bool, osmNamespace string, cfg configurator.Configurator, certManager certificate.Manager) *Server {
 	server := Server{
 		catalog: meshCatalog,
-		xdsHandlers: map[envoy.TypeURI]func(catalog.MeshCataloger, *envoy.Proxy, *xds_discovery.DiscoveryRequest, configurator.Configurator, certificate.Manager) (*xds_discovery.DiscoveryResponse, error){
+		xdsHandlers: map[envoy.TypeURI]func(catalog.MeshCataloger, *envoy.Proxy, *xds_discovery.DiscoveryRequest, configurator.Configurator, certificate.Manager) ([]types.Resource, error){
```

Comment:
I worry about the ambiguity of `[]types.Resource`. At first glance I learned that:
 - we have a list of things
 - they are in a package called types
 - they are a resource

Should this be something like `[]xds_cache.DiscoveryResponse`?  Is there a better name to increase the readability here?

Reply:
Resources is, even by checking envoy xDS proto docs and argot, a common name to refer to the object contents of the response for a given `TypeURI`. Each Response contains a number of Resources. 

I think the misunderstanding is that the handler no longer returns the marshalled objects in a prefilled DiscoveryResponse, but instead the list of Resources related to the TypeURI the handler will handle.

If this is a clarification problem, I can further comment the code or even typify the map and function signature to express additional information.
If you think this is otherwise a design problem, let's talk offline as we could have different views of what this code should/could do.